File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        maxval = nums[0]
        cursum = nums[0]
        for i in range(1, len(nums)):
            cursum = max(cursum + nums[i], nums[i])
            maxval = max(cursum, maxval)
        return maxval

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int cursum = nums[0];
        int maxval = nums[0];
        for(int i = 1; i<nums.size(); i++)
        {
            cursum = max(nums[i], cursum + nums[i]);
            maxval = max(maxval, cursum);
        }
        return maxval;
    }
};